packages/x-date-pickers/src/internals/hooks/useField/useField.ts

```diff
@@ -141,9 +142,14 @@ export const useField = <
     });
   });
 
-  const handleInputBlur = useEventCallback((...args) => {
-    onBlur?.(...(args as []));
-    setSelectedSections(null);
+  const handleInputBlur = useEventCallback((event: React.FocusEvent<HTMLInputElement>, ...args) => {
+    const isClearButton: Boolean = Boolean(
+      event.relatedTarget?.className?.split(' ')?.includes('deleteIcon'),
+    );
```

Comment:
Nitpick: this is a very fragile check, especially if we consider that we'd want to allow changing the button or icon via slots. 🤔 
Nitpick2: There is no need to explicitly type your variable if your condition resolves as a boolean.